packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,739 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { IRandom, makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { benchmark, BenchmarkType } from "@fluid-tools/benchmark";
+import {
+	createField,
+	FieldKinds,
+	getField,
+	isUnwrappedNode,
+	namedTreeSchema,
+	singleTextCursor,
+} from "../../feature-libraries";
+import { brand, unreachableCase } from "../../util";
+import { ITestTreeProvider, TestTreeProvider } from "../utils";
+import { ISharedTree } from "../../shared-tree";
+import {
+	schemaMap,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../feature-libraries/editable-tree/mockData";
+import {
+	FieldKindIdentifier,
+	fieldSchema,
+	GlobalFieldKey,
+	JsonableTree,
+	LocalFieldKey,
+	moveToDetachedField,
+	rootFieldKey,
+	rootFieldKeySymbol,
+	SchemaData,
+	TransactionResult,
+	TreeSchemaIdentifier,
+	UpPath,
+	ValueSchema,
+} from "../../core";
+
+enum TreeShape {
+	Wide = 0,
+	Deep = 1,
+}
+
+enum TestPrimitives {
+	Number = 0,
+	Float = 1,
+	String = 2,
+	Boolean = 3,
+	Map = 4,
+}
+
+async function createSharedTrees(
+	schemaData: SchemaData,
+	data: JsonableTree[],
+	numberOfTrees = 1,
+): Promise<readonly [ITestTreeProvider, readonly ISharedTree[]]> {
+	const provider = await TestTreeProvider.create(numberOfTrees);
+	for (const tree of provider.trees) {
+		assert(tree.isAttached());
+	}
+	provider.trees[0].storedSchema.update(schemaData);
+	provider.trees[0].context.root.insertNodes(0, data.map(singleTextCursor));
+	await provider.ensureSynchronized();
+	return [provider, provider.trees];
+}
+
+const globalFieldKey: GlobalFieldKey = brand("foo");
+const localFieldKey: LocalFieldKey = brand("foo");
+const rootSchemaName: TreeSchemaIdentifier = brand("Test");
+
+function getTestSchema(fieldKind: { identifier: FieldKindIdentifier }): SchemaData {
+	const testRootNodeSchema = namedTreeSchema({
+		name: rootSchemaName,
+		localFields: {
+			[localFieldKey]: fieldSchema(fieldKind),
+		},
+		globalFields: [globalFieldKey],
+		value: ValueSchema.Serializable,
+	});
+	schemaMap.set(rootSchemaName, testRootNodeSchema);
+	schemaMap.set(dataSchema.name, dataSchema);
+	return {
+		treeSchema: schemaMap,
+		globalFieldSchema: new Map([
+			[rootFieldKey, fieldSchema(FieldKinds.optional, [rootSchemaName])],
+			[globalFieldKey, fieldSchema(fieldKind)],
+		]),
+	};
+}
+
+const dataSchema = namedTreeSchema({
+	name: brand("DataSchema"),
+	localFields: {
+		foo: fieldSchema(FieldKinds.optional),
+	},
+	value: ValueSchema.Number,
+});
+
+const wideTestSizes = [5, 100, 500];
+const deepTestSizes = [5, 10, 100];
```

Comment:
I think these sizes are numbers of nodes? Would be good to make that more clear in their names and/or add a comment clarifying it.

Reply:
changed the variable names, and added clarifying comment